src/lib.rs

```diff
@@ -163,6 +165,29 @@ impl Order {
         self.account.get(&mut self.nonce, challenge_url).await
     }
 
+    /// Poll the order with exponential backoff until in a final state
+    ///
+    /// Refresh the order state from the server for `tries` times, waiting `delay` before the
+    /// first attempt and increasing the delay by a factor of 2 for each subsequent attempt.
+    ///
+    /// Yields the [`OrderStatus`] immediately if final or after `tries` attempts.
```

Comment:
Do you think it's worth describing what statuses are considered "final"?

Reply:
Yes, added that.